### PR TITLE
stackrox: Synchronize release-3.74 CI from 3.73 and master

### PR DESCRIPTION
**Pull request by msugakov**
https://issues.redhat.com/browse/ROX-15289

This should be on hold until after 3.74.0 is released. The target date for which is 2023-02-27.

### CODE REVIEW

**Comment by msugakov** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-release-3.74-gke-postgres-upgrade-tests

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,1688 @@
+presubmits:
+  stackrox/stackrox:
+  - agent: kubernetes
+    always_run: true
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml:4]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,1688 @@
...
+      volumes:
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )gke-postgres-qa-e2e-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml:206]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,1688 @@
...
+      volumes:
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )gke-qa-e2e-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml:409]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,1688 @@
...
+      volumes:
+      - name: pull-secret
+        secret:
+          secretName: registry-pull-credentials
+      - name: result-aggregator
+        secret:
+          secretName: result-aggregator
+    trigger: (?m)^/test( | .* )gke-race-condition-qa-e2e-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml:460]:
```suggestion
    always_run: false
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
I think style-checks, go-unit*, go-postgres and ui-unit are handled in release-3.74 by GHA.
/cc @janisz

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml (hunk 1 above)
Comment (msugakov) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml]: Thanks! I forgot about this. To not run into such thing next time, I put info to Confluence 

https://docs.engineering.redhat.com/display/StackRox/Making+changes+to+OpenShift+CI#MakingchangestoOpenShiftCI-PullRequestchecklist

Is there anything else you usually check in generated files that's worth mentioning?

Comment (msugakov) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-release-3.74-presubmits.yaml]: Done.

**Comment by msugakov** [comment]
> I think style-checks, go-unit*, go-postgres and ui-unit are handled in release-3.74 by GHA.
> /cc @janisz

As of 4921bb63 commit when `release-3.74` was setup GHA may have had these jobs, but they were still present for [`master`](https://github.com/openshift/release/blob/4921bb63771ecc57542deb334f75342aa7f8347a/ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml) that's why I kept them.

**Review by gavin-stackrox — APPROVED** [review]
/lgtm

**Pull request merged**